### PR TITLE
[Google Listings & Ads] Add support for picking local images from the WebView

### LINKED ISSUE #12495: [Google Listings & Ads] Add support for picking local images from the WebView

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #12495 

### Description
This PR adds adds support for picking files from device to the Composable `WebView` in our app.
For context, we already had some support as part of this [PR](https://github.com/woocommerce/woocommerce-android/pull/9257), but this PR expands on the previous implementation to make it compatible with all file types, and enables it by default on all WebView instances.

**Important:** Reviewing 67e5560 and 46ff886 separately will make reviewing the main changes of the PR easier. 

### Steps to reproduce
1. Use `trunk`.
2. Use a website that has "Google for WooCommerce" configured.
3. Open the app, and start campaign creation.
4. On the second step, try to add some images.
5. Use "Upload files" then tap on "Select files"
6. Notice that it doesn't do anything. 

### Testing information
1. Switch to this branch.
2. Repeat 2 to 5 from above.
3. Confirm you are able to select external files.

### Images/gif
https://github.com/user-attachments/assets/bc0b0e29-b043-408c-8d93-17c61b459c83

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ComposeWebChromeClient</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/234">20.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ModalStatusBarBottomSheetLayout.kt [hunk 1]
```diff
@@ -106,12 +105,17 @@ fun ModalStatusBarBottomSheetLayout(
...
-        window.navigationBarColor = sheetBackgroundColor.toArgb()
-    } else {
-        window.navigationBarColor = originalNavigationBarColor
+
+    LaunchedEffect(sheetState.currentValue) {
+        if (sheetState.currentValue != Hidden) {
+            window.navigationBarColor = sheetBackgroundColor.toArgb()
+        } else {
+            window.navigationBarColor = originalNavigationBarColor
+        }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/ModalStatusBarBottomSheetLayout.kt:118]: This is unrelated to this PR, while moving the `findActivity` extension to be reusable, I noticed that this "side-effect" code was executed during each composition, so I updated to be called only when the property changes.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/web/WebView.kt [hunk 2]
```diff
@@ -52,7 +50,7 @@ fun WCWebView(
     captureBackPresses: Boolean = true,
     wpComAuthenticator: WPComWebViewAuthenticator? = null,
     webViewNavigator: WebViewNavigator = rememberWebViewNavigator(),
-    activityRegistry: ActivityResultRegistry? = null,
+    webChromeClient: ComposeWebChromeClient = remember { ComposeWebChromeClient() },
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/web/WebView.kt:53]: Now we don't need to pass the `activityRegistry`, the client will retrieve it by itself, but to keep giving consumers a chance to handle other improvements by themselves, we can now inject a separate instance of the `ComposeWebChromeClient`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/web/ComposeWebChromeClient.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        val contract = ActivityResultContracts.StartActivityForResult()
+        val launcher = activity.activityResultRegistry.register(
+            "WebViewChooser",
+            contract
+        ) { result ->
+            val uris = FileChooserParams.parseResult(result.resultCode, result.data)
+            filePathCallback.onReceiveValue(uris)
+        }
+
+        val intent = fileChooserParams.createIntent()
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/web/ComposeWebChromeClient.kt:42]: We are using the intent provided by the WebView implementation of the device, this would match the experience of the other browsers on the device.

**Comment by hichamboushaba** [comment]
@hafizrahman I requested your review as well since this touches on the Google Ads integration, but feel free to skip the review if you prefer so.

**Comment by hafizrahman** [comment]
Didn't have the chance to look at the code today, but I smoke-tested the feature for Google for Woo and it worked properly 👍🏼

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt [hunk 4]
```diff
@@ -105,3 +107,18 @@ val Context.physicalScreenHeightInPx: Int
...
     }
+
+/**
+ * Find the closest Activity in a given Context.
+ * Inspired by the implementation in Accompanist.
+ * https://github.com/google/accompanist/blob/3b40d2b53e117bc7ca78f0d35454af9bad25f0c5/permissions/src/main/java/com/google/accompanist/permissions/PermissionsUtil.kt#L131
+ */
+fun Context.findActivity(): Activity? {
+    var context = this
+    while (context is ContextWrapper) {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt:118]: I know this was previous used in [ModalStatusBarBottomSheetLayout](https://github.com/woocommerce/woocommerce-android/commit/46ff88696172dc67e50f97849072fc64b52b68f4#diff-7180fbf9c33a13084cd4e617d227a2fc80395d54e938c07f80f56706f7574958) but could this logic result in an infinite loop?? This is a generic extension function for `Context`. If the provided `context` doesn't happen to be an activity or have a baseContext activity in its hierarchy,  this would result in an infinite loop. 
Wdyt?

**Review by JorgeMucientes — APPROVED** [review]
Nice job @hichamboushaba everything works as expected. 

I just left a minor concern. But that logic was already in the app anyway, so its not blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt (hunk 4 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt:118]: Thanks for the question @JorgeMucientes.

As stated in the comment, this logic was copied from the Accompanist library, so I hope it's safe to use.

And also looking at the logic, I think we can be sure it will never result in an infinite loop, we will never have an infinite steps while going through the `baseContext` property, it will necessarily end with the application Context which is `ContextImpl` and not a `ContextWrapper` ([code](https://cs.android.com/android/platform/superproject/+/android14-qpr3-release:frameworks/base/core/java/android/app/ContextImpl.java;l=3330-3340?q=createAppContext)).

The part about creating the app context was fun to track 😅, but let me know if you are not convinced and we can add some safety code if needed.

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt:118]: True! At the end of the hierarchy, you'll eventually land in a non`ContextWrapper` class. So, my point didn't make sense at all 😅. 
Yeah, considering there's no option for an infinite loop to happen, the code looks safe.
